py-polars/polars/internals/expr/expr.py

```diff
@@ -6095,17 +6095,22 @@ def _prepare_alpha(
     half_life: float | None = None,
     alpha: float | None = None,
 ) -> float:
-    if com is not None and alpha is None:
-        assert com >= 0.0
-        alpha = 1.0 / (1.0 + com)
-    if span is not None and alpha is None:
-        assert span >= 1.0
-        alpha = 2.0 / (span + 1.0)
-    if half_life is not None and alpha is None:
-        assert half_life > 0.0
-        alpha = 1.0 - math.exp(-math.log(2.0) / half_life)
+    if alpha is None:
+        if com is not None:
+            assert com >= 0.0
+            alpha = 1.0 / (1.0 + com)
+        elif span is not None:
+            assert span >= 1.0
+            alpha = 2.0 / (span + 1.0)
+        elif half_life is not None:
+            assert half_life > 0.0
+            alpha = 1.0 - math.exp(-math.log(2.0) / half_life)
+
     if alpha is None:
         raise ValueError("at least one of {com, span, half_life, alpha} should be set")
+
+    if not (0 < alpha <= 1):
```

Comment:
Can we throw this error on the rust side? That ensures all front-ends are correct.